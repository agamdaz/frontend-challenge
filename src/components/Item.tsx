import React from 'react';
import { style } from 'typestyle';
import { commonCss } from "./_commonCss";

const itemList = style({
  listStyleType: 'none',
});

const itemButton = style({
  width: '100%',
  padding: commonCss.doubleSpacing,
  border: 'none',
  borderBottom: commonCss.defaultBorder,
  backgroundColor: commonCss.backgroundColor,
  fontFamily: commonCss.fontFamily,
  fontWeight: 600,
  textAlign: 'left',
  cursor: 'pointer',
  $nest: {
    '&:hover': {
      backgroundColor: commonCss.secondaryColor,
    },
    '&:focus': {
      outlineColor: 'transparent',
      outlineStyle: 'none',
    },
  }
});

type ListItemProps = {
  label: string,
  onClick: (label: string) => void,
}

export const ListItem = ({label, onClick}: ListItemProps) => (
  <li data-testid="list-item" className={itemList}>
    <button className={itemButton} onClick={() => onClick(label)}>{label}</button>
  </li>
);
