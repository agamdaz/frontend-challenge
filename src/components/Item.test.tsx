import React from 'react';
import { render, screen } from '@testing-library/react';
import { ListItem } from "./Item";

test('Renders ListItem component', () => {
  render(<ListItem label="AnyItemLabel" onClick={jest.isMockFunction}/>);
  expect(screen.getByTestId('list-item')).toMatchSnapshot();
});
