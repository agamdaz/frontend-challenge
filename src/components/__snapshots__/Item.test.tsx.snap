// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ListItem component 1`] = `
<li
  class="f1v5ywig"
  data-testid="list-item"
>
  <button
    class="f1h46qw"
  >
    AnyItemLabel
  </button>
</li>
`;
